ui/src/components/mobileservices/MobileServiceView.js

```diff
@@ -33,12 +34,23 @@ class MobileServiceView extends Component {
     if (this.props.unboundServices) {
       rows.push(<h2 key="unbound-services">Unbound Services</h2>);
       this.props.unboundServices.forEach((service) => {
+        this.setDefaultBindingProperties(service);
         rows.push(<UnboundServiceRow key={service.serviceId} service={service} />);
       });
     }
     return rows;
   }
 
+  setDefaultBindingProperties(service) {
+    try {
+      if (service.bindingSchema.properties["CLIENT_ID"]) {
+        service.bindingSchema.properties["CLIENT_ID"].default = this.props.appName;
```

Comment:
is this field still showing up in the UI? Should we just hide it given that user should not need to change it anyway?

Reply:
The answer to this question is long and complicated.  

The field is required by the binding schema provided in the APB.  Even though we can prefill the information, I don't feel right about hiding what we are doing from the user.  We might instead remove the client id property from the binding schema and set that as part of creating the binding on the server side, but I don't know what knock on effects that will cause.

I am fine with making the field not editable.

Reply:
if it's not editable, and the fact that the app name is showing up on the screen, I do think there is not too much value to show this field. I still think we should make it not visible. It will keep the binging process clean.

Reply:
Oh new info, we can't remove the field because it is used by openshift's binding UI.
![image](https://user-images.githubusercontent.com/2413816/48273044-0e932700-e40e-11e8-9674-35d915165a45.png)

Reply:
![image](https://user-images.githubusercontent.com/2413816/48273071-1bb01600-e40e-11e8-8518-50e2d630e7f4.png)

Reply:
My "I don't think it is a good idea" has just been upgraded to "doing this is a terrible idea"